frontend/src/app/productListing.tsx

```diff
@@ -1,137 +1,142 @@
 import { Redirect } from 'react-router-dom'
 
-class ProductListing extends Component {
- 
-  public state = {
-    allProducts: [],
-    initialLoad: true,
-    input: '',
-    isEmptyResults: false,
-    isOpen: false,
-    isProductDetails: false,
-    loggedinStatus: false,
-    login: false,
-    productName: '',
-    redirect: false,
-    results: []
-  };
+class ProductListing extends Component<any, any, any> {
 
-  // render method transforms the react components into DOM nodes for the browser.
-  public render() {
-    const id = 'userID';
-    const descriptionKey = "description";
-    const nameKey = "name";
-    const propsKey = "match"
-    const propsKeyChild = "isExact"
-    const isOpenKey = "isOpen"
-
-    if (!this.state.loggedinStatus && this.state.initialLoad===true) {
+  constructor(props) {
+    super(props);
+    this.state = {
+      allProducts: [],
+      initialLoad: true,
+      input: '',
+      isEmptyResults: false,
+      isOpen: false,
+      isProductDetails: false,
+      loggedinStatus: false,
+      login: false,
+      productName: '',
+      redirect: false,
+      results: []
+    };
+  }
+
+  public componentDidMount() {
+
+    if (!this.state.loggedinStatus && this.state.initialLoad === true) {
       fetch("/system/sling/info.sessionInfo.json")
         .then(response => response.json())
         .then(responseJSON => {
-          if (responseJSON[id] !== 'anonymous') {
+          if (responseJSON.userID !== 'anonymous') {
             this.setState({ loggedinStatus: true })
           }
         })
     }
-   
-    if (this.props[propsKey] !== undefined) {
-      
+
+    if (this.props.match !== undefined) {
+
       // prop will be true if it comes through nav links
-       if(this.props[propsKey][propsKeyChild] === true){
-         this.state.results.map(data => {
-             (data[isOpenKey] as any) = false
-         });
-         this.setState({isProductDetails: false})
-       }
-  
-       // setting prop to false once it comes through nav links
-       this.props[propsKey][propsKeyChild]=false;
-  
+      if (this.props.match.isExact === true) {
+        this.state.results.map(data => {
+          (data.isOpen as any) = false
+        });
+        this.setState({ isProductDetails: false })
+      }
+
+      // setting prop to false once it comes through nav links
+      this.props.match.isExact = false;
+
     }
 
+    this.checkAuth()
+    this.getProducts(this.state.allProducts)
+  }
+
+  // render method transforms the react components into DOM nodes for the browser.
+  public render() {
+
+
     return (
       <React.Fragment>
-        {this.state.isProductDetails && (<ProductDetails productName={this.state.productName}/>)}
-        {this.state.initialLoad && this.getProducts(this.state.allProducts)} 
+        {this.state.isProductDetails && (<ProductDetails productName={this.state.productName} />)}
+        {/* {this.state.initialLoad && this.getProducts(this.state.allProducts)} */}
```

Comment:
Lets just remove this instead of commenting it out.

Reply:
done